### PR TITLE
Fix broken `timeout` in serial case

### PR DESCRIPTION
**Pull request by Helveg**
Does this solve the issue?

### CODE REVIEW

**Comment by drodarie** [comment]
Not sure it does: 
Tested with the file test_chunks.py from bsb-core on main branch. Currently the file contains an issue in the test (wrong import) so the test should fail in serial. Tested with the following commands in bsb-core:
`pytest tests/test_chunks.py`
`python -m unitest`
The test prints the expected error but is not considered as failed:
```
ERROR: test_default_chunk (tests.test_chunks.TestChunks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/toromis/Workspace/dbbs/bsb/bsb-test/bsb_test/parallel.py", line 90, in timed_f
    raise e
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/toromis/Workspace/dbbs/bsb/bsb-core/tests/test_chunks.py", line 20, in test_default_chunk
    self.test_single_chunk()
  File "/home/toromis/Workspace/dbbs/bsb/bsb-test/bsb_test/parallel.py", line 90, in timed_f
    raise e
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/toromis/Workspace/dbbs/bsb/bsb-core/tests/test_chunks.py", line 35, in test_single_chunk
    from bsb_test.pyconfig import cfg_single
ModuleNotFoundError: No module named 'bsb_test.pyconfig'

======================================================================
ERROR: test_single_chunk (tests.test_chunks.TestChunks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/toromis/Workspace/dbbs/bsb/bsb-test/bsb_test/parallel.py", line 90, in timed_f
    raise e
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/toromis/Workspace/dbbs/bsb/bsb-core/tests/test_chunks.py", line 35, in test_single_chunk
    from bsb_test.pyconfig import cfg_single
ModuleNotFoundError: No module named 'bsb_test.pyconfig'

```
It seems the program believes it runs in parallel?

File: bsb_test/parallel.py [hunk 1]
```diff
@@ -89,16 +89,18 @@ def timed_f(*args, **kwargs):
...
             except Exception as e:
-                import traceback, sys
-
-                errlines = traceback.format_exception(type(e), e, e.__traceback__)
-                print(
-                    *errlines,
-                    file=sys.stderr,
-                    flush=True,
-                )
                 if MPI.get_size() > 1:
```
Comment (drodarie) [inline bsb_test/parallel.py:92]: This test seems to be not sufficient to differentiate the cases in parallel and in serial.

**Review by drodarie — CHANGES_REQUESTED** [review]

**Review by drodarie — APPROVED** [review]
Works with `python -m unittest test_chunks`.
For pytest, works with the command `pytest test_chunks.py -p no:threadexception`

**Comment by Helveg** [comment]
I added a note to the README: https://github.com/dbbs-lab/bsb-test/blob/main/README.md

**Pull request merged**